### ISSUE TITLE
Refactor Object and Resource classes

### ISSUE THREAD

**Issue by benoist** [mollie/mollie-api-ruby#38]
Just to give a heads up. I will be working on refactoring the Object and Resource classes.

```ruby
class Base
  def initialize(hash)
    hash.each { |key, value|
      if value.respond_to? :each
        value = Base.new value
      end

      instance_variable_set "@#{key}", value
      self.class.send :attr_accessor, key
    }
  end
end
```

Adding the attr_accessor based on the response body can lead to memory consumption on never used attr_accessors when the response body (accidentally) returns a large list of unwanted parameters.
I will also move the CamelCase to snake_case conversion to a better location.

**Comment by Daanvm** [comment]
Great! How do you want to implement it? By defining a list of all expected properties in each `Object` or `Resource` class?

**Comment by benoist** [comment]
Not sure yet, but I'm leaning towards listing all the expected properties. This has the upside of IDE's to really see the methods. Downside is, that it requires an update of the gem when the API adds or removes properties. How often would that happen?

**Comment by Daanvm** [comment]
We do update our API from time to time. You can check our [changelog](https://www.mollie.com/en/docs/changelog) to get an idea. However, they are all small, backwards compatible changes. I don't think it's really a problem to update the list of expected properties from time to time. It would only be a small change, right?

**Comment by benoist** [comment]
Yeah it would only require a small change to add more fields. So I'll go with that option. I hopefully have some time tomorrow evening or otherwise the next one. But really hope to have something by the end of this week.

**Comment by Daanvm** [comment]
This was fixed in PR #41.
